types/address.go

```diff
@@ -239,37 +235,33 @@ func (aa AccAddress) Bytes() []byte {
 
 // AccAddress.String() is expensive and if unoptimized dominantly showed up in profiles,
 // yet has no mechanisms to trivially cache the result given that AccAddress is a []byte type.
-// With the string interning below, we are able to achieve zero cost allocations for string->[]byte
-// because the Go compiler recognizes the special case of map[string([]byte)] when used exactly
-// in that pattern. See https://github.com/cosmos/cosmos-sdk/issues/8693.
 var addMu sync.Mutex
-var addrStrMemoize = make(map[string]string)
+var addrStrMemoize, _ = simplelru.NewLRU(1000, nil)
```

Comment:
This initialization should be in an `init()` function, so that we can check the error returned and `panic()` if it is != `nil`.

Reply:
I was following the current design. That being said I agree with you - I will use `init`.